include/mbgl/map/backend_scope.hpp

```diff
@@ -0,0 +1,17 @@
+#pragma once
+
+namespace mbgl {
+
+class Backend;
+
+class BackendScope {
+public:
+    BackendScope(Backend&);
+    ~BackendScope();
+
+private:
+    Backend* priorBackend;
```

Comment:
As an additional safeguard to aid in debugging, we could make this a doubly linked list and `assert` that there is not a `nextBackend` in the destructor.